src/Network/Network/Common/ResourceIdHelper.cs

```diff
@@ -0,0 +1,29 @@
+﻿using System;
```

Comment:
@lwajswaj we have a common model object, [`ResourceIdentifier`](https://github.com/Azure/azure-powershell-common/blob/master/src/ResourceManager/Version2016_09_01/Utilities/Models/ResourceIdentifier.cs), that we tell developers to use when dealing with resource ids -- is there any reason we should introduce this class rather than using this existing class?

If we can't use the existing class, please add a license header to this file.

Reply:
I don't see this file has a method to support parsing all the different piece of a resource id from a string; I followed the approach same it's done in here: https://github.com/Azure/azure-powershell/blob/686808fc5623bf091dd2d51c5f587bc0b372b875/src/Websites/Websites/Utilities/CmdletHelpers.cs#L247; if you think I should use or extend ResourceIdentifier, I would; otherwise I'll add the disclaimer.

Reply:
Added license header

Reply:
I think it would be preferrable to use ResourceIndentifier

```csharp
if (ParameterSetName == "GetByResourceId")
{
    ResourceIdentifier identifier = new ResourceIdentifier(ResourceId);
    if (identifier.ResourceType == "Microsoft.Network/virtualNetworks/subnets")
    {
        this.Name = identifier.ResourceName;
        string virtualNetworkName = identifier.ParentResource.Substring(identifier.ParentResource.LastIndexOf('/') + 1);
        this.VirtualNetwork = this.GetVirtualNetwork(identifier.ResourceGroupName, virtualNetworkName);
    }
    else
    {
        // throw error
    }
}
```

Reply:
Changed